src/js/legend.js

```diff
@@ -0,0 +1,198 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
+        model: {
+            // dataSet accepts:
+            // 1. an array of objects. Must contain "id", "value" and "label" variables.
+            // Example: [{id: string, value: number, label: string} ... ]
+            dataSet: []
+        },
+        legendOptions: {
+            width: 300,
+            height: 300,
+            colors: null, // An array of colors for the legend generated for corresponding values of model.dataSet
+            sort: true // Whether or not to sort the data by values when creating the legend
+        },
+        styles: {
+            legend: "gpii-ca-pieChart-legend",
+            table: "gpii-ca-pieChart-legend-table",
+            row: "gpii-ca-pieChart-legend-table-row",
+            legendColorCell: "gpii-ca-pieChart-legend-table-row-legend-color-cell",
+            labelCell: "gpii-ca-pieChart-legend-table-row-legend-label-cell",
+            valueCell: "gpii-ca-pieChart-legend-table-row-legend-value-cell"
```

Comment:
Can we shorten names for cell elements? Same as these element defined in selectors block.